frontends/web/src/routes/settings/components/device-settings/show-recovery-words-setting.tsx

```diff
@@ -34,25 +36,40 @@ type TDialog = {
 const ShowRecoveryWordsSetting = ({ deviceID }: TProps) => {
   const { t } = useTranslation();
   const [inProgress, setInProgress] = useState(false);
+  const [showDialog, setShowDialog] = useState(false);
+
+  const confirmShowWords = async () => {
+    setShowDialog(false);
+    setInProgress(true);
+    await showMnemonic(deviceID);
+    setInProgress(false);
 
-  const handleShowMnemonic = () => {
-    confirmation(t('backup.showMnemonic.description'), async result => {
-      if (result) {
-        setInProgress(true);
-        await showMnemonic(deviceID);
-        setInProgress(false);
-      }
-    });
   };
   return (
     <>
       <SettingsItem
         settingName={t('backup.showMnemonic.title')}
         secondaryText={t('deviceSettings.backups.showRecoveryWords.description')}
         extraComponent={<ChevronRightDark />}
-        onClick={handleShowMnemonic}
+        onClick={() => setShowDialog(true)}
       />
       <ShowMnemonicWaitDialog inProgress={inProgress} />
+
+      <Dialog title={t('backup.showMnemonic.title')} open={showDialog} onClose={() => setShowDialog(false)}>
```

Comment:
the `onClose` does not seem necessary but otherwise you can't open the dialogue again after exiting with `esc` key instead of the cancel button.